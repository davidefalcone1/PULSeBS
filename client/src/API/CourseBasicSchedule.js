class CourseBasicSchedule{
    constructor(id, courseId, day, startTime, endTime, classroom) {
        this.id = id;
        this.courseId = courseId;
        this.day = day;
        this.startTime = startTime;
        this.endTime = endTime;
        this.classroom = classroom;
    }

    static fromJson(json){
        const temp =  Object.assign(new CourseBasicSchedule(), json);
        return temp;
    }
}

export default CourseBasicSchedule;