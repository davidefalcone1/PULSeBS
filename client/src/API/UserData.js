class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

    static fromJson(json){
        const temp =  Object.assign(new UserData(), json);
        return temp;
    }
}

export default UserData;