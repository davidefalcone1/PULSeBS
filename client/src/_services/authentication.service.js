import { BehaviorSubject } from 'rxjs';


const currentUserSubject = new BehaviorSubject(JSON.parse(localStorage.getItem('user')));

export const authenticationService = {
    login,
    logout,
    user: currentUserSubject.asObservable(),
    get currentUserValue () { return currentUserSubject.value }
};

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if ([401, 403].indexOf(response.status) !== -1) {
                // auto logout if 401 Unauthorized or 403 Forbidden response returned from api
                authenticationService.logout();
            }

            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
}

function login(username, password) {
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ username, password })
    };
    return new Promise(async function (resolve, reject) {
        try {
            const response = await fetch('/users/authenticate', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ username, password }),
            });
            const user = await response.json().then(handleResponse);
            if (response.ok) {
                localStorage.setItem('user', JSON.stringify(user));
                currentUserSubject.next(user);
                resolve(user);
            }
            else
                reject();
        }
        catch (e) {
            reject();
        }
    });

}
function logout() {

    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
    };
    localStorage.removeItem('user');
    currentUserSubject.next(null);
    return fetch('/logout',requestOptions);
}