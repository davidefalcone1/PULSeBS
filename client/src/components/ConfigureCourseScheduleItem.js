import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import Button from 'react-bootstrap/Button';

const configureCourseScheduleItem = (props) => {

  return (
    <ListGroup.Item id = {props.schedule.id + "-" + props.schedule.scheduleId}>
        <div className="d-flex w-100 pt-3 justify-content-between no-gutters">
            <DayField id = {props.schedule.id} day = {props.schedule.day}/>
            <StartTimeField id = {props.schedule.id} startTime = {props.schedule.startTime}/>
            <EndTimeField id = {props.schedule.id} endTime = {props.schedule.endTime}/>
            <ButtonField id = {props.schedule.id} edit = {props.editCourseSchedule}
                delete = {props.deleteCourseSchedule}/>
        </div>
    </ListGroup.Item>
  );
}

function DayField(props){
    return(
        <div className="col-sm-3">
            <h4 id={"dayOfSchedule" + props.id}>
                {props.day}
            </h4>
        </div>
    );
}

function StartTimeField(props){
    return(
        <div className="col-sm-3">
            <p id={"sTimeOfSchedule" + props.id}>
                {props.startTime}
            </p>
        </div>
    );
}

function EndTimeField(props){
    return(
        <div className="col-sm-3">
            <p id={"eTimeOfSchedule" + props.id}>
                {props.endTime}
            </p>
        </div>
    );
}
function ButtonField(props){
    return(
        <div className="col-sm-4">
            <Button variant="info" onClick={(event) => {
                event.preventDefault();
                props.edit();
            }} id={"editButtonFieldOflesson" + props.id}>
                Edit
            </Button>
            <Button variant="danger" onClick={(event) => {
                event.preventDefault();
                props.delete(props.id);
            }} id={"deleteButtonFieldOflesson" + props.id}>
                Delete
            </Button>
        </div>
    );
}

export default configureCourseScheduleItem;