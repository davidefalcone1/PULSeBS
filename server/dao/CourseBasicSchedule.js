class CourseBasicSchedule{
    constructor(id, courseId, day, startTime, endTime, classroom) {
        this.id = id;
        this.courseId = courseId;
        this.day = day;
        this.startTime = startTime;
        this.endTime = endTime;
        this.classroom = classroom;
    }
}

module.exports = CourseBasicSchedule;