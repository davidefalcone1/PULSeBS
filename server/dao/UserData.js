class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

}

module.exports = UserData;