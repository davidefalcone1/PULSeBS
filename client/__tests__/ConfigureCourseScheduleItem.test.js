import React from 'react';
import renderer from 'react-test-renderer';
import { AuthContext } from '../src/_services/AuthContext';
import ConfigureCourseScheduleItem from '../src/components/ConfigureCourseScheduleItem';
import { BrowserRouter } from 'react-router-dom';
import CourseBasicSchedule from '../src/API/CourseBasicSchedule';
import UserData from '../src/API/UserData';
import moment from 'moment';

describe('ConfigureCoursesItem', ()=>{
    test('Renders correctly', ()=>{
        const component = renderer.create(
            <BrowserRouter>
                <AuthContext.Provider value={{user: new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')}}>
                    <ConfigureCourseScheduleItem schedule={new CourseBasicSchedule(1, 1, 'Mon', moment('2020-09-08T14:30').format('HH:mm'), moment('2020-09-08T17:00').format('HH:mm'), 'A1')} editCourseSchedule={()=>{}} deleteCourseSchedule={()=>{}}/>
                </AuthContext.Provider>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});