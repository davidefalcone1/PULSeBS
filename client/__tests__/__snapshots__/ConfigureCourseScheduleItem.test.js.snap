// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureCoursesItem Renders correctly 1`] = `
<div
  className="list-group-item"
  disabled={false}
  id="1-undefined"
  onClick={[Function]}
>
  <div
    className="d-flex w-100 pt-3 justify-content-between no-gutters"
  >
    <div
      className="col-sm-3"
    >
      <h4
        id="dayOfSchedule1"
      >
        Mon
      </h4>
    </div>
    <div
      className="col-sm-2"
    >
      <p
        id="sTimeOfSchedule1"
      >
        14:30
      </p>
    </div>
    <div
      className="col-sm-2"
    >
      <p
        id="eTimeOfSchedule1"
      >
        17:00
      </p>
    </div>
    <div
      className="col-sm-2"
    >
      <h4
        id="classroomOfSchedule1"
      >
        A1
      </h4>
    </div>
    <div
      className="col-sm-3"
    >
      <button
        className="btn btn-info"
        disabled={false}
        id="editButtonFieldOflesson1"
        onClick={[Function]}
        type="button"
      >
        Edit
      </button>
       
      <button
        className="btn btn-danger"
        disabled={false}
        id="deleteButtonFieldOflesson1"
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
